tests/phpunit/CRM/Contact/Import/Parser/ContactTest.php

```diff
@@ -1519,4 +1537,32 @@ protected function importCSV(string $csv, array $mapper, array $submittedValues
     $form->postProcess();
   }
 
+  /**
+   * @param string $csv
+   * @param array $mapper
+   * @param string $field
+   *
+   * @throws \API_Exception
+   * @throws \CRM_Core_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
+   */
+  private function validateMultiRowCsv(string $csv, array $mapper, string $field): void {
```

Comment:
Since the assertion here seems new+important, I figured I'd kick the tires on it. If I randomly throw in bad data, it complains. So that's good. :)

I'm not sure I understand the signature. `$csv` makes sense as filename. What are `$mapper` and `$field`? (Docblock would be good - but doesn't have to be on this PR specifically.)

Reply:
mapper is pretty integral to the concept of these import forms - it's documented in quite a few places - I can perhaps find a 'see'. I'm starting to think `$field` might get removed again in later tests as I start to build on it.